django/utils/cache.py

```diff
@@ -328,15 +327,15 @@ def _generate_cache_key(request, method, headerlist, key_prefix):
         value = request.META.get(header)
         if value is not None:
             ctx.update(value.encode())
-    url = hashlib.md5(iri_to_uri(request.build_absolute_uri()).encode('ascii'))
+    url = hashlib.md5(request.build_absolute_uri().encode('ascii'))
     cache_key = 'views.decorators.cache.cache_page.%s.%s.%s.%s' % (
         key_prefix, method, url.hexdigest(), ctx.hexdigest())
     return _i18n_cache_key_suffix(request, cache_key)
 
 
 def _generate_cache_header_key(key_prefix, request):
     """Return a cache key for the header cache."""
-    url = hashlib.md5(iri_to_uri(request.build_absolute_uri()).encode('ascii'))
+    url = hashlib.md5(request.build_absolute_uri().encode('ascii'))
```

Comment:
Interesting.

This is definitely the thing to do since `build_absolute_uri` already returns an _uri_ but I assume the removal of the double escaping might generate a new key. Not a big deal but maybe something worthy of a release note.

Reply:
@charettes I don't think so, to the best of my knowledge `iri_to_uri` is idempotent. Especially since `%` is part of the safe characters it will not double escape escaped characters. I'd be happy to be proven wrong though as that would indicate that I have a gross misunderstanding about how `iri_to_uri` works and after all this function is sometimes security-relevant. So please double check and see if you can come up with any iri that results in a different uri when `iri_to_uri` is applied a second time.

Reply:
@apollo13 you're completely right. Look I missed the inclusion of `'%'` in `safe` somehow.